### ISSUE TITLE
godot-zig seems to crash because of double precision

### ISSUE THREAD

**Issue by fire** [godot-zig/godot-zig#5]
Context:

I was evaluating using zig with libgodot https://github.com/migeran/libgodot_project to provide bindings for elixir https://github.com/E-xyza/zigler

Without the elixir integration, you can directly achieve your goal of zig running Godot Engine through libgodot with the zig code being the entrypoint into the operating system.

Description:

![image](https://github.com/godot-zig/godot-zig/assets/32321/cfceffe5-9bdf-4308-8abf-6a91ce953680)

When I click on the buttons Godot Engine crashes.

```
% godot -e --path ./project
Godot Engine v4.3.dev.custom_build - https://godotengine.org
Vulkan 1.2.275 - Forward+ - Using Device #0: Apple - Apple M2 Pro

Godot Engine v4.3.dev.custom_build - https://godotengine.org
Vulkan 1.2.275 - Forward+ - Using Device #0: Apple - Apple M2 Pro

onResized
Hello GodotZig!
onTimeout
on_item_focused: idx = 1

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.3.dev.custom_build
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x000000018c713584 _sigtramp + 56
[2] 2   libexample.dylib                    0x0000000112eea8d0 ExampleNode.on_item_focused + 96
[3] 3   libexample.dylib                    0x0000000112eea8d0 ExampleNode.on_item_focused + 96
[4] 4   libexample.dylib                    0x0000000112eeabd4 api.Godot.MethodBinderT(fn (*ExampleNode, i64) void).bind_call + 368
[5] GDExtensionMethodBind::call(Object*, Variant const**, int, Callable::CallError&) const (in godot) (gdextension.cpp:214)
[6] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (in godot) (object.cpp:837)
[7] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (in godot) (callable.cpp:0)
[8] Object::emit_signalp(StringName const&, Variant const**, int) (in godot) (object.cpp:1202)
[9] Error Object::emit_signal<int>(StringName const&, int) (in godot) (object.h:933)
[10] ItemList::gui_input(Ref<InputEvent> const&) (in godot) (item_list.cpp:732)
[11] Control::_call_gui_input(Ref<InputEvent> const&) (in godot) (control.cpp:0)
[12] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (in godot) (viewport.cpp:1538)
[13] Viewport::_gui_input_event(Ref<InputEvent>) (in godot) (viewport.cpp:1768)
[14] Viewport::push_input(Ref<InputEvent> const&, bool) (in godot) (viewport.cpp:3229)
[15] Window::_window_input(Ref<InputEvent> const&) (in godot) (window.cpp:0)
[16] void call_with_variant_args_helper<Window, Ref<InputEvent> const&, 0ul>(Window*, void (Window::*)(Ref<InputEvent> const&), Variant const**, Callable::CallError&, IndexSequence<0ul>) (in godot) (binder_common.h:304)
[17] Variant Callable::call<Ref<InputEvent>>(Ref<InputEvent>) const (in godot) (variant.h:856)
[18] DisplayServerMacOS::_dispatch_input_event(Ref<InputEvent> const&) (in godot) (display_server_macos.mm:386)
[19] Input::_parse_input_event_impl(Ref<InputEvent> const&, bool) (in godot) (input.cpp:773)
[20] Input::flush_buffered_events() (in godot) (input.cpp:1044)
[21] DisplayServerMacOS::process_events() (in godot) (display_server_macos.mm:3025)
[22] OS_MacOS::run() (in godot) (os_macos.mm:776)
[23] main (in godot) (godot_main_macos.mm:89)
[24] 24  dyld                                0x000000018c35a0e0 start + 2360
-- END OF BACKTRACE --
```

I'm using a godot 4.3 master customized and with precision doubles. Let me know if you need more info. I'm missing some but I don't know what.

**Comment by fire** [comment]
I suspect that the godot-zig isn't generating the doubles bindings correctly.

**Comment by elvisxzhou** [comment]
Have no idea about your issue, could you please provide a test case, or shed some light on how to reproduce this?

**Comment by elvisxzhou** [comment]
fixed in 7d26d883db00dfc0f4b642b6b3f6fc7142811fc7
zig build -Dprecision=double do the trick
